### PR TITLE
Update bundles per arm mixin usage

### PR DESCRIPTION
**Pull request by vdice**
# What does this change
Updates/refactors all example and workshop bundles that had previously used the deprecated `azure` mixin to use the `arm` mixin.

In doing so and testing the subsequent bundles, I also sometimes made some tweaks to other aspects of the bundles, for instance, updating terraform and terraform provider versions (and corresponding usage), updating some ARM template values, aligning parameter name formats (usually updating hyphenated params to snake case as the majority dictated), etc.

# What issue does it fix
Closes https://github.com/deislabs/porter/issues/861

# Notes for the reviewer
Please call out any changes/updates that we don't wish to make at this time.

# Checklist
- [ ] Unit Tests
- [ ] Documentation
  - [ ] Documentation Not Impacted

### CODE REVIEW

File: docs/content/wiring.md [hunk 1]
```diff
@@ -31,8 +31,8 @@ Once a parameter has been declared in the `porter.yaml`, Porter provides a simpl
 
 ```yaml
 install:
-- description: "Install MySQL"
-  helm:
+- helm:
```
Comment (carolynvs) [inline docs/content/wiring.md]: Dang I thought I had gotten all of these, so glad you noticed! 🙇‍♀

File: examples/aks-spring-music/README.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ The bundle will use the service principal created above to interact with Azure.
     ```
 
 * Update params for your deployment
-    * change the `invocationImage` Docker repo to match your Docker Hub account (line 4)
+    * change the `tag` Docker repo to match your Docker Hub account (line 5)
```
Comment (carolynvs) [inline examples/aks-spring-music/README.md]: nit/suggestion: leave off the line # as it's easy to drift without us noticing

File: examples/aks-spring-music/porter.yaml [hunk 3]
```diff
@@ -35,6 +36,15 @@ install:
         - "--tenant" 
         - "{{ bundle.credentials.TENANT_ID}}"
 
+  - exec:
```
Comment (carolynvs) [inline examples/aks-spring-music/porter.yaml]: @jeremyrickard Now that we have the az mixin, long term (not this PR) do we want to update this to use the az mixin?

File: examples/azure-wordpress/params.ini [hunk 4]
```diff
@@ -1,2 +1,2 @@
-mysql_password=ilovecnab
-server_name=ilovecnab
\ No newline at end of file
+mysql_password=iloveporter-127c6!J9$
```
Comment (carolynvs) [inline examples/azure-wordpress/params.ini]: lol strong stance there 😉 Take that CNAB

File: examples/azure-wordpress/porter.yaml [hunk 5]
```diff
@@ -67,8 +72,8 @@ install:
         externalDatabase.database: "{{ bundle.parameters.database_name }}"
 
 uninstall:
-  # TODO: enable once the porter-azure mixin supports this action
-  # - azure:
+  # TODO: enable once the porter-arm mixin supports this action
```
Comment (carolynvs) [inline examples/azure-wordpress/porter.yaml]: arm supports uninstall where the old azure mixin didn't. Maybe a follow-up would be to implement the stanza for uninstall

File: workshop/porter-tf-aci/azure/bundle/terraform/mysql.tf [hunk 6]
```diff
@@ -1,42 +1,37 @@
 resource "random_string" "password" {
   length = 16
   special = true
-  override_special = "/@\" "
+  override_special = "/@£$"
```
Comment (carolynvs) [inline workshop/porter-tf-aci/azure/bundle/terraform/mysql.tf]: Why did this change?

File: examples/azure-wordpress/porter.yaml (hunk 5 above)
Comment (vdice) [reply examples/azure-wordpress/porter.yaml]: Oh nice!  I'd be down to add it in this same PR, if you are?

File: examples/azure-wordpress/params.ini (hunk 4 above)
Comment (vdice) [reply examples/azure-wordpress/params.ini]: 😂 Also added those extry chars b/c Azure was like 'password not complex enuff'

File: workshop/porter-tf-aci/azure/bundle/terraform/mysql.tf (hunk 6 above)
Comment (vdice) [reply workshop/porter-tf-aci/azure/bundle/terraform/mysql.tf]: 😂 the `"` was messing with my VSCode formatting... all lines below it weren't write... 'twas thinking it was all in one big quote.  Obvi more of an extension issue buuut I figured the actual override chars didn't really matter (pulled the new chars from the azurerm docs/examples).

File: examples/aks-spring-music/README.md (hunk 2 above)
Comment (vdice) [reply examples/aks-spring-music/README.md]: Good call; will remove.

File: examples/azure-wordpress/porter.yaml (hunk 5 above)
Comment (vdice) [reply examples/azure-wordpress/porter.yaml]: Oops, n/m.  Not quite ready for primetime: https://github.com/deislabs/porter-arm/blob/master/pkg/arm/uninstall.go

Comment (vdice) [reply examples/azure-wordpress/porter.yaml]: Added a link to the issue around implementing uninstall (https://github.com/deislabs/porter-arm/issues/7) in the comment.

Comment (carolynvs) [reply examples/azure-wordpress/porter.yaml]: 🤦‍♀

File: workshop/porter-tf-aci/azure/bundle/terraform/mysql.tf (hunk 6 above)
Comment (carolynvs) [reply workshop/porter-tf-aci/azure/bundle/terraform/mysql.tf]: 🖖 Carry on!

**Pull request merged**